### PR TITLE
Add arguments to communities.list for geographic range.

### PR DESCRIPTION
**Pull request by ntdkhiem**
####  Summary / Highlights
Added extra field  `distance` to `community.location` when `zipcode` argument is provided to `communities.list` to display the distance between given zipcode and community's zipcode. An optional `max_distance` with a default value of 25 miles will filter communities with a distance less than or equal to that value.  

#### Requirements (place an `x` in each `[ ]`)
* [X] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [X] I've tested my changes manually.
* [X] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [X] I've given my PR a meaningful title.
* [X] I linked this PR to the relevant issue.
* [X] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.

### CODE REVIEW

**Review by BradHN1 — COMMENTED** [review]
@ntdkhiem - I suspect you have an auto-formatter on python files, which changes spaces, quote marks, tabs and such.  This makes it difficult for me to review changes since many lines can change unintentionally.  And your auto-formatter may have different rules than someone else's.  If possible see if you can disable that.

**Comment by ntdkhiem** [comment]
Thank you for letting me know. I converted back to the original format with my changes. I think it should just display only my changes now.

**Comment by ntdkhiem** [comment]
> @ntdkhiem - I suspect you have an auto-formatter on python files, which changes spaces, quote marks, tabs and such. This makes it difficult for me to review changes since many lines can change unintentionally. And your auto-formatter may have different rules than someone else's. If possible see if you can disable that.

^

File: src/api/store/community.py [hunk 1]
```diff
@@ -586,16 +615,70 @@ def list_communities(
     ) -> Tuple[list, MassEnergizeAPIError]:
         try:
             if context.is_sandbox:
-                communities = Community.objects.filter(
-                    is_deleted=False, is_approved=True
-                ).exclude(subdomain="template").order_by('name')
+                communities = (
+                    Community.objects.filter(is_deleted=False, is_approved=True)
```
Comment (BradHN1) [inline src/api/store/community.py:619]: add is_published=True

**Review by BradHN1 — APPROVED** [review]
LGTM - great work!

**Pull request merged**